stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1082,6 +1082,43 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         
     }
     
+    // slightly faster paths for common sequences
+    
+    public init<S: Sequence>(_ elements: S) where S.Iterator.Element == UInt8 {
+        let underestimatedCount = elements.underestimatedCount
+        self.init(count: underestimatedCount)
+        var idx = 0
+        for byte in elements {
```

Comment:
If instead of hand rolling the loop you call `initialize(from:)` on your underlying pointer then types like `Array` will memcpy directly into your buffer, which should be even faster.